src/workflows/auth/definition.ts

```diff
@@ -1,11 +1,25 @@
-import { IProfile } from 'src/entities';
+import { IProfile, IRequest } from 'src/entities';
 import { Struct } from 'src/structs';
 export type Auth =
-  Struct<'loginWithGoogle', {
-    success: (profile: IProfile) => void;
-    error?: ErrorCallback;
-  }>
-  | Struct<'logOut', {
-    success: VoidCallback;
-    error: ErrorCallback;
-  }>
+  | Struct<
+      'loginWithGoogle',
+      {
+        success: (profile: IProfile) => void;
+        error?: ErrorCallback;
+      }
+    >
+  | Struct<
+      'logOut',
+      {
+        success: VoidCallback;
+        error: ErrorCallback;
+      }
+    >
+  | Struct<
+      'request',
```

Comment:
You may want to rename this to something meaningful since in this context request maybe in noun form.

```suggestion
      'createRequest',
```

Because later on we will have `approveRequest`, and `rejectRequest`

Reply:
understood sir. didn't realized that